File: fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/error/DoubleOpenedRollCallException.java
package com.github.dedis.popstellar.utility.error;

import com.github.dedis.popstellar.R;

/**
 * An exception that indicates the roll call cannot be opened as there's already another one in
 * progress
 */
public class DoubleOpenedRollCallException extends GenericException {

  public DoubleOpenedRollCallException(String id) {
    super("Impossible to open roll call " + id + " as another roll call is still open");
  }

  @Override
  public int getUserMessage() {
    return R.string.already_open_roll_call_exception;
  }

  @Override
  public Object[] getUserMessageArguments() {
    return new Object[0];
  }
}

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/error/DoubleOpenedRollCallException.kt
package com.github.dedis.popstellar.utility.error

import com.github.dedis.popstellar.R

/**
 * An exception that indicates the roll call cannot be opened as there's already another one in
 * progress
 */
class DoubleOpenedRollCallException(id: String) :
    GenericException("Impossible to open roll call id $id as another roll call is still open") {
    override val userMessage: Int
        get() = R.string.already_open_roll_call_exception
    override val userMessageArguments: Array<Any?>
        get() = arrayOfNulls(0)
}